### PR TITLE
gstreamer1*, gupnp* and friends: update to current versions

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

@mascguy @catap This will probably not build on CI straightaway, however everything does build for me locally. Please help me with some testing and finalizing portfiles. It should work, perhaps with minor amendments.

Hopefully this will be helpful for Kirill’s work on `webkit2-gtk`.

Two caveats:
1. I did not test Qt5 variant (one port has it as the default, I just made it conditional to systems where the current Qt5 builds, according to port details).
2. I did not test universal builds and for now removed the old code (which won’t work anyway, since makefiles switched to meson).

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
> This is amazing job. Is it possible to delivery it commit by commit?

@catap Do you mean a separate PR for each port/commit? This will not work, since they depend on each other, and builds will fail with archaic versions we got now. (Or otherwise we need to think through the sequence and merge every one PR before submitting the next.)
Or if you mean to make every minor change to each port in a separate commit, that is doable in principle, but a) PR gonna become a mess and b) to be honest, this thing already took many hours (and will still require more to make required changes so that everything builds on CI), I would rather not add up to that something non-functional.

**Comment by barracuda156** [comment]
@catap @mascguy From look of things it may build fine now.

Portfiles still need to be reviewed though.

**Comment by barracuda156** [comment]
@mascguy @catap Provisionally this can be tested now.

P. S. I did some cleanup with old copy-pasted Python settings and changed `gobject-introspection` to path style, also made `gettext` (wherever it was in dependencies) a build dep, while its runtime a lib dep. If anything of these do not work, I will revert/correct it.
But before the clean-up commit all CI already passed.

**Comment by catap** [comment]
Maybe add `gtk4` variant?

**Comment by barracuda156** [comment]
@catap Could you help with that? I do not want to add anything blindly, and I assume `gtk4` won’t build – though I have not tried it recently.

**Comment by catap** [comment]
> @catap Could you help with that? I do not want to add anything blindly, and I assume `gtk4` won’t build – though I have not tried it recently.

a bit later

File: net/gupnp-av/Portfile [hunk 1]
```diff
@@ -2,51 +2,43 @@
 
 PortSystem          1.0
 PortGroup           compiler_blacklist_versions 1.0
-PortGroup           gobject_introspection 1.0
+PortGroup			meson 1.0
```
Comment (mascguy) [inline net/gupnp-av/Portfile:5]: Minor nit-pick: Replace the tabs on this line, in favor of spaces.

File: gnome/gstreamer1-gst-plugins-ugly/Portfile [hunk 2]
```diff
@@ -1,67 +1,66 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
+PortGroup			meson 1.0
```
Comment (mascguy) [inline gnome/gstreamer1-gst-plugins-ugly/Portfile:4]: Replace tabs with spaces, on this line.

File: gnome/gstreamer1-gst-plugins-good/Portfile [hunk 3]
```diff
@@ -26,146 +27,131 @@ master_sites        https://gstreamer.freedesktop.org/src/${my_name}/
...
-# <rdar://problem/9694837> - Fixed in Xcode 4.2 (clang 211.10.1)
-# blacklist compilers that don't support C11 typedef redefinition
-# gstflvmux.h:73: error: redefinition of typedef ‘GstFlvMuxPadClass’
-compiler.c_standard 2011
+default_variants            +pulseaudio +x11 +gtk3
 
 livecheck.type      regex
 livecheck.url       ${master_sites}
 livecheck.regex     "${my_name}-(\\d+\\\.\\d*\[02468\](?:\\.\\d+)*)${extract.suffix}"
+
```
Comment (mascguy) [inline gnome/gstreamer1-gst-plugins-good/Portfile:157]: Remove this trailing newline.

Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-good/Portfile]: Done, this and tabs. Thank you for pointing out.

**Comment by mascguy** [comment]
This is definitely great stuff, and would love to see this merged.

That being said, we'll ultimately need to rev-bump all dependents too [as part of this PR]. And we need to be sure that downstream components still work, as these are very foundational.

I can start by testing `json-glib` with various dependents, given that I maintain a number of those.

But let's defer the dependent rev-bumps for the moment, until we're reasonably confident that everything here is good.

**Comment by mascguy** [comment]
For `json-glib`, you need to add build dep `gettext`, and lib dep `gettext-runtime`.

**Comment by mascguy** [comment]
Sergey, go ahead and create a separate PR, for `json-glib` alone. (Keep the update in here too for now, if needed for the other updates.)

Reason being, the latest version appears to retain ABI compatibility, based on local testing. So we might as well get that updated first, and simplify this PR.

**Comment by mascguy** [comment]
> Sergey, go ahead and create a separate PR, for `json-glib` alone. (Keep the update in here too for now, if needed for the other updates.)
> 
> Reason being, the latest version appears to retain ABI compatibility, based on local testing. So we might as well get that updated first, and simplify this PR.

On second thought, I've taken co-ownership of that port, so I'll go ahead and update. (Though I'll be sure to credit you for the update, when committing.)

**Comment by mascguy** [comment]
> > Sergey, go ahead and create a separate PR, for `json-glib` alone. (Keep the update in here too for now, if needed for the other updates.)
> 
> On second thought, I've taken co-ownership of that port, so I'll go ahead and update. (Though I'll be sure to credit you for the update, when committing.)

Rebased to drop the update for `json-glib`. (Latter now available via master.)

**Comment by catap** [comment]
@mascguy, @barracuda156, @Gcenx and everyone here: I'm working on update webkit2-gtk. It doesn't (yet) finished. But at least it has been built on my machine.

You may track the progress here https://github.com/catap/macports-ports/tree/webkit2-gtk-devel and if you would like to help => welcome!

**Comment by barracuda156** [comment]
@mascguy @Gcenx If someone can add needed changes for 10.8+ directly into the PR, that would be great (these are not my ports, feel free to modify however needed). Since locally I test on 10.6 for ppc, I can verify what works or does not there, and any conditional change applied to later systems won’t break old ones anyway.

Otherwise I can pick needed changes from my end, but I need to know what is required. Should I just borrow whatever missing here from @Gcenx branch?

**Comment by barracuda156** [comment]
@catap I would have tried, but even patches fail to apply:
```
--->  Checksumming webkitgtk-2.40.5.tar.xz
--->  Extracting webkit2-gtk-devel
--->  Extracting webkitgtk-2.40.5.tar.xz
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work" && /opt/local/bin/xz -dc '/opt/local/var/macports/distfiles/webkit-gtk/webkitgtk-2.40.5.tar.xz' | /usr/bin/gnutar --no-same-owner -xf - 
--->  Applying patches to webkit2-gtk-devel
--->  Applying patch-webkit2gtk-macports.diff
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-webkit2gtk-macports.diff'
patching file Source/JavaScriptCore/shell/PlatformGTK.cmake
patching file Source/WebCore/CMakeLists.txt
patching file Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
patching file Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp
--->  Applying patch-webkit2gtk-272-macports.diff
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-webkit2gtk-272-macports.diff'
patching file Source/WebKit/NetworkProcess/NetworkLoadParameters.h
Hunk #1 succeeded at 38 with fuzz 1 (offset 3 lines).
patching file Source/WebKit/Platform/IPC/ArgumentCoders.h
Hunk #1 succeeded at 53 with fuzz 2 (offset 19 lines).
patching file Source/WebKit/Shared/SandboxExtension.h
Hunk #1 succeeded at 34 (offset 1 line).
--->  Applying patch-webkit2gtk-2282-unprotected-egl-changes.diff
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-webkit2gtk-2282-unprotected-egl-changes.diff'
patching file Source/WebCore/platform/graphics/gstreamer/GRefPtrGStreamer.cpp
Hunk #2 succeeded at 578 (offset 70 lines).
Hunk #3 succeeded at 596 with fuzz 2 (offset 70 lines).
patching file Source/WebCore/platform/graphics/gstreamer/GRefPtrGStreamer.h
--->  Applying patch-webkit2gtk-2282-os-log-availability.diff
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-webkit2gtk-2282-os-log-availability.diff'
patching file Source/ThirdParty/ANGLE/src/common/debug.cpp
Hunk #2 succeeded at 261 (offset 29 lines).
--->  Applying patch-enable-plugin-architecture-unix.diff
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-enable-plugin-architecture-unix.diff'
patching file Source/WebKit/config.h
--->  Applying patch-bundle-link-webcore.diff
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-bundle-link-webcore.diff'
patching file Source/WebKit/PlatformGTK.cmake
--->  Applying patch-darwin-GTK.diff
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-darwin-GTK.diff'
patching file Source/JavaScriptCore/API/WebKitAvailability.h
patching file Source/WebKit/Platform/IPC/ArgumentCoders.h
Hunk #1 succeeded at 43 with fuzz 1.
patching file Source/WebKit/Platform/IPC/Attachment.h
patching file Source/WebKit/Platform/IPC/Connection.h
patching file Source/WebKit/Platform/SharedMemory.h
patching file Source/WebKit/PlatformGTK.cmake
patching file Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp
patching file Source/bmalloc/bmalloc/ProcessCheck.h
--->  Applying patch-gstreamer-before-1.18.diff
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-gstreamer-before-1.18.diff'
patching file Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
--->  Applying patch-ramsize.diff
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-ramsize.diff'
patching file Source/WTF/wtf/RAMSize.cpp
Hunk #1 succeeded at 28 with fuzz 2 (offset -1 lines).
Hunk #2 succeeded at 48 (offset 1 line).
Hunk #3 succeeded at 58 (offset 1 line).
Hunk #4 succeeded at 103 (offset 5 lines).
--->  Applying patch-source-wtf-wtf-unix-cputimeunix-cpp-darwin-version-restore.diff
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-source-wtf-wtf-unix-cputimeunix-cpp-darwin-version-restore.diff'
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git Source/WTF/wtf/unix/CPUTimeUnix.cpp Source/WTF/wtf/unix/CPUTimeUnix.cpp
|index 27990893..fe279865 100644
|--- Source/WTF/wtf/unix/CPUTimeUnix.cpp
|+++ Source/WTF/wtf/unix/CPUTimeUnix.cpp
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.
2 out of 2 hunks ignored
Command failed:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/work/webkitgtk-2.40.5" && /usr/bin/patch -p0 < '/opt/PPCSnowLeopardPorts/www/webkit2-gtk-devel/files/patch-source-wtf-wtf-unix-cputimeunix-cpp-darwin-version-restore.diff'
Exit code: 1
Error: Failed to patch webkit2-gtk-devel: command execution failed
Error: See /opt/local/var/macports/logs/_opt_PPCSnowLeopardPorts_www_webkit2-gtk-devel/webkit2-gtk-devel/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there is a bug.
Error: Processing of port webkit2-gtk-devel failed
```

**Comment by catap** [comment]
@barracuda156 I haven't ported old patches yet, it still requires tons of work be built on on old system; thus modern system isn't (yet) ready.

**Comment by barracuda156** [comment]
@mascguy Updated `gstreamer1*` ports to 1.22.6, rebased to master.

**Comment by barracuda156** [comment]
@mascguy CI fail on another port, `gupnp-av`:
```
  FAILED: libgupnp-av/libgupnp-av-1.0.3.dylib.p/gupnp-didl-lite-parser.c.o 
  /usr/bin/clang -Ilibgupnp-av/libgupnp-av-1.0.3.dylib.p -Ilibgupnp-av -I../gupnp-av-0.14.1/libgupnp-av -Iinternal -I../gupnp-av-0.14.1/internal -I/opt/local/include/glib-2.0 -I/opt/local/lib/glib-2.0/include -I/opt/local/include/libxml2 -I/opt/local/include -fcolor-diagnostics -Wall -Winvalid-pch -std=c11 -O0 -g -Werror=deprecated-declarations -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -arch x86_64 -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -Wpointer-arith -Wmissing-declarations -Wformat=2 -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Wbad-function-cast -Wold-style-definition -Wunused -Wuninitialized -Wshadow -Wmissing-noreturn -Wmissing-format-attribute -Wredundant-decls -Werror=implicit -Werror=nonnull -Werror=init-self -Werror=main -Werror=missing-braces -Werror=sequence-point -Werror=return-type -Werror=trigraphs -Werror=array-bounds -Werror=write-strings -Werror=address -Werror=int-to-pointer-cast -Werror=pointer-to-int-cast -fno-strict-aliasing -Wno-int-conversion -MD -MQ libgupnp-av/libgupnp-av-1.0.3.dylib.p/gupnp-didl-lite-parser.c.o -MF libgupnp-av/libgupnp-av-1.0.3.dylib.p/gupnp-didl-lite-parser.c.o.d -o libgupnp-av/libgupnp-av-1.0.3.dylib.p/gupnp-didl-lite-parser.c.o -c ../gupnp-av-0.14.1/libgupnp-av/gupnp-didl-lite-parser.c
  ../gupnp-av-0.14.1/libgupnp-av/gupnp-didl-lite-parser.c:233:15: error: 'xmlRecoverMemory' is deprecated [-Werror,-Wdeprecated-declarations]
          doc = xmlRecoverMemory (didl, strlen (didl));
                ^
```
Not sure what has broken with it, I did not touch it now. All `gstreamer1*` updates built fine.

**Comment by barracuda156** [comment]
@mascguy Please let me know when you test this locally.

I will look into what happened with gupnp* stuff which built earlier.

**Comment by barracuda156** [comment]
@mascguy Ok, this should be fixable:
```
  ../gupnp-av-0.14.1/libgupnp-av/gupnp-didl-lite-parser.c:233:15: error: 'xmlRecoverMemory' is deprecated [-Werror,-Wdeprecated-declarations]
          doc = xmlRecoverMemory (didl, strlen (didl));
                ^
  /opt/local/include/libxml2/libxml/parser.h:870:1: note: 'xmlRecoverMemory' has been explicitly marked deprecated here
  XML_DEPRECATED
  ^
  /opt/local/include/libxml2/libxml/xmlversion.h:447:43: note: expanded from macro 'XML_DEPRECATED'
  #    define XML_DEPRECATED __attribute__((deprecated))
                                            ^
  1 error generated.
```

**Comment by barracuda156** [comment]
@Schamschula @mascguy Hmm, what happened to this one now?
```
  Failed to parse file python/py-pyerfa/Portfile with subport 'py37-pyerfa': can't read "github.author": no such variable
  Failed to parse file python/py-pyerfa/Portfile with subport 'py38-pyerfa': can't read "github.author": no such variable
```

**Comment by Schamschula** [comment]
Oops! I accidentally kept some code I copied from a different port with pinned a version for older Python subports.

Should be fixed in https://github.com/macports/macports-ports/commit/f6f88d7d8184c5333c5273ba6002fb84edbba36b

**Comment by barracuda156** [comment]
@Schamschula Looks like CI are working again, thanks.

**Comment by barracuda156** [comment]
@mascguy Okay, now everything builds fine.

**Comment by barracuda156** [comment]
@mascguy Is anything needed from my side for this?

**Comment by mascguy** [comment]
No, per my last comment, I need to thoroughly test with all of this everywhere. And that's going to take considerable time.

Patience please...

**Comment by mascguy** [comment]
> Just incase this gets resurrected there's now a `gl-headers` package.

Good to know, thanks for the heads-up Dean!

**Pull request closed without merging**